diff -ur qt-everywhere-opensource-src-4.7.0-tp/src/corelib/kernel/qmetaobject_p.h qt-everywhere-opensource-src-4.7.0-tp-QTBUG-9354/src/corelib/kernel/qmetaobject_p.h
--- qt-everywhere-opensource-src-4.7.0-tp/src/corelib/kernel/qmetaobject_p.h	2010-03-10 14:56:51.000000000 +0100
+++ qt-everywhere-opensource-src-4.7.0-tp-QTBUG-9354/src/corelib/kernel/qmetaobject_p.h	2010-04-04 00:14:37.000000000 +0200
@@ -287,7 +287,8 @@
         }
 
         // cv qualifers can appear after the type as well
-        if (t != e && (e - t >= 5 && strncmp("const", t, 5) == 0)) {
+        if (!is_ident_char(c) && t != e && (e - t >= 5 && strncmp("const", t, 5) == 0)
+            && (e - t == 5 || !is_ident_char(t[5]))) {
             t += 5;
             while (t != e && is_space(*t))
                 ++t;
