diff -up qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qapplication.cpp.grass qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qapplication.cpp
diff -up qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qguieventdispatcher_glib.cpp.grass qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qguieventdispatcher_glib.cpp
--- qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qguieventdispatcher_glib.cpp.grass	2010-06-01 21:03:15.000000000 -0500
+++ qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qguieventdispatcher_glib.cpp	2010-06-29 12:07:58.545073715 -0500
@@ -177,6 +177,8 @@ QGuiEventDispatcherGlibPrivate::QGuiEven
     memset(&x11EventSource->pollfd, 0, sizeof(GPollFD));
     x11EventSource->flags = QEventLoop::AllEvents;
     x11EventSource->q = 0;
+    // workaround NULL dereference https://bugzilla.redhat.com/show_bug.cgi?id=498111#c59 -- Rex
+    if (x11EventSource->d)
     x11EventSource->d = 0;
 
     g_source_attach(&x11EventSource->source, mainContext);
