diff -up qt-everywhere-opensource-src-4.7.0-beta1/src/gui/kernel/qapplication.cpp.QT_GRAPHICSSYSTEM qt-everywhere-opensource-src-4.7.0-beta1/src/gui/kernel/qapplication.cpp
--- qt-everywhere-opensource-src-4.7.0-beta1/src/gui/kernel/qapplication.cpp.QT_GRAPHICSSYSTEM	2010-05-03 19:43:24.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.0-beta1/src/gui/kernel/qapplication.cpp	2010-05-17 13:56:33.851069964 -0500
@@ -774,6 +774,10 @@ void QApplicationPrivate::construct(
 
     qt_is_gui_used = (qt_appType != QApplication::Tty);
     process_cmdline();
+    // the environment variable has the lowest precedence of runtime graphicssystem switches
+    if (graphics_system_name.isEmpty()) {
+        graphics_system_name = QString::fromLocal8Bit(qgetenv("QT_GRAPHICSSYSTEM"));
+    }
     // Must be called before initialize()
     qt_init(this, qt_appType
 #ifdef Q_WS_X11
