diff -up qt-everywhere-opensource-src-4.8.0-tp/src/network/network.pro.openssl qt-everywhere-opensource-src-4.8.0-tp/src/network/network.pro
--- qt-everywhere-opensource-src-4.8.0-tp/src/network/network.pro.openssl	2011-05-23 05:26:21.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.0-tp/src/network/network.pro	2011-06-01 14:25:55.012307358 -0500
@@ -20,6 +20,8 @@ include(access/access.pri)
 include(bearer/bearer.pri)
 include(kernel/kernel.pri)
 include(socket/socket.pri)
+## hack around typedef mismatch for SSL_CTX_ctrl (with and without const) -- Rex
+QMAKE_CXXFLAGS += -fpermissive
 include(ssl/ssl.pri)
 
 QMAKE_LIBS += $$QMAKE_LIBS_NETWORK
diff -up qt-everywhere-opensource-src-4.8.0-tp/src/network/ssl/qsslsocket_openssl.cpp.openssl qt-everywhere-opensource-src-4.8.0-tp/src/network/ssl/qsslsocket_openssl.cpp
diff -up qt-everywhere-opensource-src-4.8.0-tp/src/network/ssl/qsslsocket_openssl_symbols.cpp.openssl qt-everywhere-opensource-src-4.8.0-tp/src/network/ssl/qsslsocket_openssl_symbols.cpp
diff -up qt-everywhere-opensource-src-4.8.0-tp/src/network/ssl/qsslsocket_openssl_symbols_p.h.openssl qt-everywhere-opensource-src-4.8.0-tp/src/network/ssl/qsslsocket_openssl_symbols_p.h
