diff -ur qt-x11-opensource-src-4.3.4/src/network/qsslsocket_openssl_symbols.cpp qt-x11-opensource-src-4.3.4-openssl/src/network/qsslsocket_openssl_symbols.cpp
--- qt-x11-opensource-src-4.3.4/src/network/qsslsocket_openssl_symbols.cpp	2008-02-19 18:47:39.000000000 +0100
+++ qt-x11-opensource-src-4.3.4-openssl/src/network/qsslsocket_openssl_symbols.cpp	2008-04-02 04:55:24.000000000 +0200
@@ -254,7 +254,7 @@
         return false;
     }
 #else
-    QLibrary libssl(QLatin1String("ssl"));
+    QLibrary libssl(QLatin1String("libssl.so." SHLIB_VERSION_NUMBER));
     if (!libssl.load()) {
         // Cannot find libssl
         qWarning("QSslSocket: cannot find ssl library: %s.",
@@ -262,7 +262,7 @@
         return false;
     }
 
-    QLibrary libcrypto(QLatin1String("crypto"));
+    QLibrary libcrypto(QLatin1String("libcrypto.so." SHLIB_VERSION_NUMBER));
     if (!libcrypto.load()) {
         // Cannot find libcrypto
         qWarning("QSslSocket: cannot find crypto library: %s.",
