diff -ur qt-x11-opensource-src-4.5.0/src/gui/text/qfontdatabase_x11.cpp qt-x11-opensource-src-4.5.0-pa-in/src/gui/text/qfontdatabase_x11.cpp
--- qt-x11-opensource-src-4.5.0/src/gui/text/qfontdatabase_x11.cpp	2009-02-25 22:09:23.000000000 +0100
+++ qt-x11-opensource-src-4.5.0-pa-in/src/gui/text/qfontdatabase_x11.cpp	2009-05-07 03:21:08.000000000 +0200
@@ -818,7 +818,7 @@
     "div", // Thaana
     "hi", // Devanagari
     "bn", // Bengali
-    "pa", // Gurmukhi
+    "pa-in", // Gurmukhi
     "gu", // Gujarati
     "or", // Oriya
     "ta", // Tamil
@@ -883,7 +883,7 @@
     "div", // Thaana
     "hi",  // Devanagari
     "bn",  // Bengali
-    "pa",  // Gurmukhi
+    "pa-in",  // Gurmukhi
     "gu",  // Gujarati
     "or",  // Oriya
     "ta",  // Tamil
