--- qt-x11-opensource-src-4.5.1/configure	2009-04-22 01:57:35.000000000 +0200
+++ qt-x11-opensource-src-4.5.1.new/configure	2009-06-02 17:58:00.000000000 +0200
@@ -3970,7 +3984,7 @@
     [ "$CFG_DEV" = "yes" ] && SYNCQT_OPTS="$SYNCQT_OPTS -check-includes"
     if [ "$OPT_SHADOW" = "yes" ]; then
         "$outpath/bin/syncqt" $SYNCQT_OPTS
-    elif [ "$CFG_DEV" = "yes" ]; then
+    elif [ "$CFG_DEV" = "yes" ] || [ ! -d $relpath/include ]; then
         QTDIR="$relpath" perl "$outpath/bin/syncqt" $SYNCQT_OPTS
     fi
 fi
