diff -up qt-x11-opensource-src-4.5.0/src/gui/text/qfontengine_ft.cpp.disable_ft_lcdfilter qt-x11-opensource-src-4.5.0/src/gui/text/qfontengine_ft.cpp
--- qt-x11-opensource-src-4.5.0/src/gui/text/qfontengine_ft.cpp.disable_ft_lcdfilter	2009-04-14 13:56:07.000000000 -0500
+++ qt-x11-opensource-src-4.5.0/src/gui/text/qfontengine_ft.cpp	2009-04-14 13:59:00.000000000 -0500
@@ -72,9 +72,9 @@
 #include FT_CONFIG_OPTIONS_H
 #endif
 
-#if defined(FT_LCD_FILTER_H) && defined(FT_CONFIG_OPTION_SUBPIXEL_RENDERING)
-#define QT_USE_FREETYPE_LCDFILTER
-#endif
+//#if defined(FT_LCD_FILTER_H) && defined(FT_CONFIG_OPTION_SUBPIXEL_RENDERING)
+//#define QT_USE_FREETYPE_LCDFILTER
+//#endif
 
 #ifdef QT_LINUXBASE
 #include FT_ERRORS_H
