--- qt/src/gui/kernel/qwidget.cpp	2008-08-01 16:08:30.000000000 +0100
+++ qt/src/gui/kernel/qwidget.cpp	2008-08-05 11:22:40.000000000 +0100
@@ -9095,8 +9095,6 @@
         break;
 #endif
     case Qt::WA_NativeWindow:
-        if (!qApp->testAttribute(Qt::AA_DontCreateNativeWidgetSiblings) && parentWidget())
-            parentWidget()->d_func()->enforceNativeChildren();
         if (on && !internalWinId() && testAttribute(Qt::WA_WState_Created))
             d->createWinId();
         break;
