diff -up qt-x11-opensource-src-4.5.0-rc1/src/network/kernel/qhostaddress.h.qhostaddress qt-x11-opensource-src-4.5.0-rc1/src/network/kernel/qhostaddress.h
--- qt-x11-opensource-src-4.5.0-rc1/src/network/kernel/qhostaddress.h.qhostaddress	2009-02-04 10:05:29.000000000 -0600
+++ qt-x11-opensource-src-4.5.0-rc1/src/network/kernel/qhostaddress.h	2009-02-22 13:32:08.000000000 -0600
@@ -37,6 +37,7 @@
 #ifndef QHOSTADDRESS_H
 #define QHOSTADDRESS_H
 
+#include <QtCore/qpair.h>
 #include <QtCore/qstring.h>
 #include <QtNetwork/qabstractsocket.h>
 
